### PR TITLE
test: siftest verify

### PR DESCRIPTION
**Pull request by timlind**

### CODE REVIEW

File: cmd/siftest/main.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+func VerifyOpenLong(clientCtx client.Context,
+	from string,
+	height int64,
+	collateralAmount sdk.Uint,
+	collateralAsset,
+	borrowAsset string,
+	leverage sdk.Dec) error {
+
+	return nil
+}
```
Comment (pgoos) [inline cmd/siftest/main.go:175]: is it actually needed if it returns `nil` ?

File: cmd/siftest/readme.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+```shell
+verifying removal...
+
+Wallet native balance before 499999807448701559072997868307527
+Wallet external balance before 499999999999999999999022148651615
+
+Wallet native balance after 499999807448702459100223367291740 
+Wallet external balance after 499999999999999999999022148656694 
+
+Wallet native diff 900027225498984213 (expected: 1000032419169645384 unexpected: -100005193670661171)
```
Comment (pgoos) [inline cmd/siftest/readme.md:74]: do we want to show in the examples a failure case (think this is what `unexpected` states)?

File: cmd/siftest/readme.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+Pool native liabilities before 0
+Pool external liabilities before 500000000000
+Pool native depth (including liabilities) before 499999999999999003000000496
+Pool external depth (including liabilities) before 500000000000001000000000000
+
+Pool health after 0.999999999999999000
+Pool native custody after 0
+Pool external custody after 0
+Pool native liabilities after 0
+Pool external liabilities after 0
```
Comment (pgoos) [inline cmd/siftest/readme.md:129]: What if there are multiple MTPs and we're only closing one of them? How are we verifying the values then?

File: cmd/siftest/test.go [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+	accountNumber, seq, err := txf.AccountRetriever().GetAccountNumberSequence(clientCtx, key.GetAddress())
+	if err != nil {
+		panic(err)
+	}
+
+	//txf = txf.WithAccountNumber(accountNumber).WithSequence(seq)
+	//err = TestAddLiquidity(clientCtx, txf, key)
+	//if err != nil {
+	//	panic(err)
+	//}
```
Comment (pgoos) [inline cmd/siftest/test.go:41]: shall we get rid of this code?

File: cmd/siftest/test.go [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+
+	fmt.Printf("Pool sent diff: %s\n", poolNativeDiff.String())
+	fmt.Printf("Pool received diff: %s\n", poolExternalDiff.String())
+	fmt.Printf("Address received diff: %s\n", rowanDiff.String())
+	fmt.Printf("Address sent diff: %s\n", cethDiff.String())
+
+	return nil
+}
+
+func TestOpenPosition(clientCtx client.Context, txf tx.Factory, key keyring.Info) error {
```
Comment (pgoos) [inline cmd/siftest/test.go:216]: Naming convention question: why some functions are prefixed `Test` and some `GetVerify` ?

File: cmd/siftest/verify.go [hunk 6]
```diff
@@ -0,0 +1,583 @@
...
+		externalDiff.Sub(sdk.NewIntFromBigInt(externalAmount.BigInt()).Neg()))
+
+	//fmt.Printf("External deduction %s \n", externalDiff.String())
+	//fmt.Printf("External expected %s \n\n", externalAmount.String())
+	//
+	//fmt.Printf("Native diff %s \n", nativeDiff.String())
+	//fmt.Printf("Native expected %s \n", sdk.NewIntFromBigInt(nativeAmount.BigInt()).Neg().String())
+	//fmt.Printf("Native diff - expected %s \n\n", nativeDiff.Sub(sdk.NewIntFromBigInt(nativeAmount.BigInt()).Neg()).String())
+
+	//fmt.Printf("LP units expected diff %s \n", lpUnits.String())
```
Comment (pgoos) [inline cmd/siftest/verify.go:181]: Shall we remove this code?

File: cmd/siftest/readme.md (hunk 3 above)
Comment (timlind) [reply cmd/siftest/readme.md:129]: Added expected values.

File: cmd/siftest/readme.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+Return amount: 494008999461
+Loss: 0
+
+Wallet collateral balance after: 488999999999999994008999412 (diff: 494008999412)
+Wallet custody balance after: 499999211271873732838129967124882 (diff: -99999999999999996)
```
[the remark this answers is not in the document]
Comment (timlind) [reply cmd/siftest/readme.md:135]: Added collateral and custody assets so it's easier to see which are fees, as well as expected diff for collateral balance (return amount).

File: cmd/siftest/test.go (hunk 5 above)
Comment (timlind) [reply cmd/siftest/test.go:210]: The Test functions execute transactions, they aren't in use yet.

Comment (pgoos) [reply cmd/siftest/test.go:210]: Would be great to execute transactions and verify them ina single go. Are you actively working on it or what is the plan?

File: cmd/siftest/main.go (hunk 1 above)
Comment (timlind) [reply cmd/siftest/main.go:146]: keeping this here as WIP

**Review by canercandan — APPROVED** [review]
LGTM

**Pull request merged**